Heroes.ReplayParser/ParseOptions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Heroes.ReplayParser
+{
+    /// <summary>
+    /// Parse options, represented as a class.
+    /// 
+    /// Default options are automatically set, typical use would
+    /// be to either use one of the provided static option sets,
+    /// or to set your own option set in the initializer
+    /// 
+    /// i.e. new ParseOption() {
+    /// ShouldParseUnits = true
+    /// }
+    /// </summary>
+    public class ParseOptions
+    {
+        public bool IgnoreErrors { get; set; } = false;
+        public bool AllowPTR { get; set; } = false;
+        public bool ShouldParseEvents { get; set; } = true;
+        public bool ShouldParseUnits { get; set; } = false;
+        public bool ShouldParseMouseEvents { get; set; } = false;
+        public bool ShouldParseDetailedBattleLobby { get; set; } = true;
+        public bool ShouldParseMessageEvents { get; set; } = false;
+        public bool ShouldParseStatistics { get; set; } = true;
```

Comment:
Looks good overall :) Yes I've needed an options class like this for awhile

One request: Could we set the defaults to match what the current defaults are, to hopefully avoid some confusion.  Users can of course specify whatever they want

The change would be to enable 'ShouldParseUnits', enable 'ShouldParseMouseEvents', and disable 'ShouldParseDetailedBattleLobby'